motorlib/motor.py

```diff
@@ -246,6 +247,11 @@ def runSimulation(self, callback=None):
             _, _, gamma, _, _ = self.propellant.getCombustionProperties(pressure)
             exitPressure = self.nozzle.getExitPressure(gamma, pressure)
             simRes.channels['exitPressure'].addData(exitPressure)
+            
+            if (exitPressure <= (self.nozzle.getSeparationPressureRatio() * self.config.getProperty('ambPressure'))) and flowSeparationWarning == False:
```

Comment:
It is technically a bit less efficient but cleaner in my opinion if this lives with the mass flux and pressure checks down below outside of this loop. We don't need the `flowSeparationWarning` variable if this is moved. If looping over all of the simulation results to find the max or min pressure again at the end of the simulation is ever a performance problem, the use of methods like `getPeakMassFlux()` instead of just calling max on the data would easily allow the object calculate the max while items are being added to speed it back up.